fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -3116,12 +3116,12 @@ public void createTableAsSelect(CreateTableAsSelectStmt stmt) throws DdlExceptio
                 }
                 TypeDef typeDef;
                 Expr resultExpr = resultExprs.get(i);
-                // varchar/char transfer to string
-                if (resultExpr.getType().isStringType()) {
+                if (resultExpr.getType().isStringType() && resultExpr.getType().getLength() < 0) {
```

Comment:
Why adding `resultExpr.getType().getLength() < 0` condition?
If this is a bug, please described in issue.

Reply:
-1 means that we don't know the varchar length ,so we transfer it to string, but if we know the length of string, just keep it same as before